app/controllers/transactions_controller.rb

```diff
@@ -73,6 +73,6 @@ def transaction_params
   end
 
   def valid_account?(account)
-    account && account.user_id == current_user.id
+    account && current_user.accounts.ids.include?(account.id)
```

Comment:
refaz o teste, se ainda for possivel cadastrar a implementação esta errada.

<img width="1850" alt="Captura de Tela 2024-09-06 às 01 05 09" src="https://github.com/user-attachments/assets/bda777e9-a849-4b87-ae62-62ecccd7bcd0">

Reply:
sim, foi possivel.
qual a diferença da forma que voce fez para a forma que eu fiz?
no qual o teste funcionou 

![image](https://github.com/user-attachments/assets/43f4986f-718f-4e6f-aec6-eff481321986)

Reply:
vc insiste em ser um dev front. 
em qual camada vc acha que deve ficar as validações?